### PR TITLE
executor: escape the escape character in show create table

### LINKED ISSUE #28908: 'show create table' statement‘s default value should take escape character into account, like '\'.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
create table test(a int primary key, b varchar(20) default '\\');
show create table test;
```

### 2. What did you expect to see? (Required)

sql create statement’s default value takes escape character into account. 
```sql
CREATE TABLE `test` (
  `a` int(11) DEFAULT NULL,
  `b` varchar(20) DEFAULT '\\'
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
```

### 3. What did you see instead (Required)
sql create statement‘s default value didn't take escape character into account.
```sql
CREATE TABLE `test` (
  `a` int(11) DEFAULT NULL,
  `b` varchar(20) DEFAULT '\'
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
```

### 4. What is your TiDB version? (Required)

5.7.25-TiDB-v5.2.1

### PR DESCRIPTION
**Pull request by fanrenhoo**
Signed-off-by: fanrenhoo <fanrenhoo@sina.com>

### What problem does this PR solve?

Issue Number: close #28908

Problem Summary: "show create table" statement‘s default value display didn't take escape character into account

### What is changed and how it works?
in util/format/format.go, the func "replace" add '\\' format replace case
and also add a test case for this in executor/show_test.go
tested and it works fine

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
Please check the linter.

**Comment by fanrenhoo** [comment]
the previous committed pull shown failed test in CI, I checked, its testify's format_test.go reported. after investigation, i believe the fix is right, but the previous test case not right: if use `\n\r\000` to test OutputFormat, the replace logic(from '\n’ to "\\n", '\r' to "\\r", '\000' to "\\0") will not be called at all, the test will not reach the aim. So I just rewrite the test case to meet its original purpose: display with escape charactor taken into account

File: util/format/format_test.go [hunk 1]
```diff
@@ -47,6 +48,7 @@ z
 	expect = "abc3%e x y z\n "
 	checkFormat(t, f, buf, str, expect)
 
-	str2 := OutputFormat(`\'\000abc\n\rdef`)
```
Comment (xiongjiwei) [inline util/format/format_test.go:50]: the [back quote](https://go.dev/ref/spec#:~:text=character%20sequences%20between-,back%20quotes,-%2C%20as%20in%20%60foo) is a better way to write the case, no need to change this case

File: util/format/format_test.go [hunk 2]
```diff
@@ -47,6 +48,7 @@ z
...
-	assert.Equal(t, str2, "\\''\\000abc\\n\\rdef")
+	str1 := fmt.Sprintf("%c%c%s%c%c%s", '\'', '\000', "abc", '\n', '\r', "def")
+	str2 := OutputFormat(str1)
```
Comment (xiongjiwei) [inline util/format/format_test.go:52]: the [back quote](https://go.dev/ref/spec#:~:text=character%20sequences%20between-,back%20quotes,-%2C%20as%20in%20%60foo) is a better way to write the case, no need to change this case

**Comment by xiongjiwei** [comment]
the linter will check the code format, and it report
```
[](https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check/detail/tidb_ghpr_check/49640/pipeline#step-77-log-68)[2022-02-12T05:49:20.172Z] executor/show_test.go

[2022-02-12T05:49:20.172Z] make: *** [fmt] Error 1
```
please make sure you run `make check` before you submit your commit

**Comment by fanrenhoo** [comment]
> the linter will check the code format, and it report
> 
> ```
> [2022-02-12T05:49:20.172Z] executor/show_test.go
> [2022-02-12T05:49:20.172Z] make: *** [fmt] Error 1
> ```
> 
> please make sure you run `make check` before you submit your commit

make check the correct code format for show_test.go and commit, please check

File: util/format/format_test.go (hunk 2 above)
Comment (fanrenhoo) [reply util/format/format_test.go:52]: make check linter and correct code format for show_test.go and commit again, please check. And for the "format_test.go", sure back quote will be better and display right.  but what i confusing is in the old test case, like `\n`, `\r` will be treated as '\' 'r' '\' 'n' seperately, the func "replace" will not be used at all, so if meet a char '\n', will not act like back quote test does.  Anyway, if sure the previous format_test.go testcase is right, so just ignore this file's change

Comment (xiongjiwei) [reply util/format/format_test.go:52]: get it

**Review by xiongjiwei — APPROVED** [review]

**Comment by fanrenhoo** [comment]
Hi, this seem stay a long time, could please take time review, THKS much.

**Review by hawkingrei — APPROVED** [review]

**Comment by xiongjiwei** [comment]
@fanrenhoo seems the test is not passed, PTAL

**Comment by fanrenhoo** [comment]
> @fanrenhoo seems the test is not passed, PTAL

Hi, I saw this. But i'm confused that the testutil is used everywhere in whole TestShowCreateTable func, I just copied and modified, and in my native machine I run test again seem right, and code format also seem ok. Any hint?

**Comment by fanrenhoo** [comment]
> @fanrenhoo seems the test is not passed, PTAL

Hi, I saw this. But i'm confused that the testutil is used everywhere in whole TestShowCreateTable func, I just copied and modified, and in my native machine I run test again seem right, and code format also seem ok. Any hint?

**Comment by xiongjiwei** [comment]
testutil is renamed to testkit, you need to update your branch to see the newest changes

**Comment by fanrenhoo** [comment]
> testutil is renamed to testkit, you need to update your branch to see the newest changes

Ok, got it. I modified with the newest base and commited again, please check

**Comment by xiongjiwei** [comment]
still have failed test

**Comment by fanrenhoo** [comment]
> still have failed test

The fix should be right, error actually not relate with this issue. i rebase the new update and now it should be ok. THKS

**Pull request merged**